### ISSUE TITLE
[DEV] 

### ISSUE THREAD

**Issue by Tommatheussen** [flixlix/energy-period-selector-plus#36]
I wanted to take a look at getting the following issues fixed:
- https://github.com/flixlix/energy-period-selector-plus/issues/35
- https://github.com/flixlix/energy-period-selector-plus/issues/21
- https://github.com/flixlix/energy-period-selector-plus/issues/20
- https://github.com/flixlix/energy-period-selector-plus/issues/18

I think I found the problem why these are happening, I see this card is requesting energy data with the start date offset by 1 hour (at least for me): `"start_time":"2024-02-18T22:00:00.000Z"`, compared to the stock date picker (which sends`"start_time":"2024-02-18T23:00:00.000Z"`. I suspect that is why the stock cards don't properly render, as they receive data from overlapping days/timeperiods?

When looking at where and how the data gets loaded, I see in https://github.com/flixlix/energy-period-selector-plus/blob/main/src/energy/index.ts that a lot of the functions are duplicated or reimplemented. Is there a reason why we can't import these helper functions from HA itself?

**Comment by shiner66** [comment]
I don't really believe this repo is being actively maintaned, you might want to give it a shot yourself(?)

**Comment by flixlix** [comment]
@shiner66 yeah, unfortunately I've not had the time to properly maintain it in the last couple of months. I'll try to start fixing issues in the upcoming weeks

**Comment by shiner66** [comment]
Thank you!

Sent with Spark
